corehq/apps/domain/decorators.py

```diff
@@ -380,6 +380,7 @@ def _inner(request, *args, **kwargs):
                 DIGEST: login_or_digest_ex(allow_cc_users=True),
                 API_KEY: login_or_api_key_ex(allow_cc_users=True),
                 FORMPLAYER: login_or_formplayer_ex(allow_cc_users=True),
+                OAUTH2: login_or_oauth2_ex(allow_cc_users=True),
```

Comment:
nit: seems like we could use `get_auth_decorator_map` to create the map here if we just add the `allow_cc_users` param to  `get_auth_decorator_map`

Reply:
the other thing that would do is add `FORMPLAYER` auth to our Tastypie APIs. is that something we want? (I wasn't sure what it was for, which was why I didn't bother, but agree it'd be nice to deduplicate that code).

Reply:
I don't think there's any harm in it since it would only be activated if the appropriate header was there. I think right now it will 500 if you add the HMAC header which is probably not what we want either.

Reply:
added in https://github.com/dimagi/commcare-hq/pull/29367/commits/074a2737a75fff0fe0d5a312a2ee50b225bc1f90 and https://github.com/dimagi/commcare-hq/pull/29367/commits/23f9740502ef552d275a86c633013bf7c3b3739f

Reply:
and https://github.com/dimagi/commcare-hq/pull/29367/commits/a0a07426e7927f209c1ac5a372c238cd507539c7